### PR TITLE
Report xorb compression size

### PR DESCRIPTION
**Pull request by seanses**
Fix BAC-162

Tests:
1. Under a repo with history before this, add a csv file of 373909 bytes
```
di@di-mbp ~/tt/pyxet-build % git xet merkledb cas-stat
{
"total_cas_bytes" : 559694603,
"total_file_bytes" : 559917404
}
di@di-mbp ~/tt/pyxet-build % ls -l
total 1062232
-rw-r--r--@ 1 di  staff     373909 May  6 14:22 Stocks_data (1).csv
-rw-r--r--  1 di  staff  527042656 May  6 14:22 rpyxet.abi3.so
-rw-r--r--  1 di  staff   16437430 May  6 14:22 xet-linux-arm64.tar.gz
di@di-mbp ~/tt/pyxet-build % git add .
Git-Xet 0.13.4: Processing data: 16.03 MiB | 16.03 MiB/s, done.
16.03 MiB added, stored 110.65 KiB (99.3% reduction)
di@di-mbp ~/tt/pyxet-build % git commit -m "add stock data"
[main e73a7fc] add stock data
 1 file changed, 3 insertions(+)
 create mode 100644 Stocks_data (1).csv
di@di-mbp ~/tt/pyxet-build % git push >/dev/null 2>&1
di@di-mbp ~/tt/pyxet-build % git xet merkledb cas-stat
{
"total_compressed_cas_bytes" : 559807907,
"total_cas_bytes" : 560068512,
"total_file_bytes" : 560291313
}
```
2. Under a new repo that commits a csv file of 76775041 bytes after this
```
di@di-mbp ~/tt/comp % ls -l
total 149952
-rw-r--r--  1 di  staff  76775041 May  6 13:56 sometrain.csv
di@di-mbp ~/tt/comp % git add sometrain.csv
Git-Xet 0.13.4: Processing data: 73.22 MiB | 29.15 MiB/s, done.
73.22 MiB added, stored 14.82 MiB (79.8% reduction)
di@di-mbp ~/tt/comp % git commit -m "add a csv"
[main 0887633] add a csv
 1 file changed, 3 insertions(+)
 create mode 100644 sometrain.csv
di@di-mbp ~/tt/comp % git push
di@di-mbp ~/tt/comp % git xet merkledb cas-stat
{
"total_compressed_cas_bytes" : 15544790,
"total_cas_bytes" : 76775041,
"total_file_bytes" : 76775041
}
```

### CODE REVIEW

File: rust/gitxetcore/src/data/mdb.rs [hunk 1]
```diff
@@ -892,11 +893,20 @@ pub async fn cas_stat_git(config: &XetConfig) -> errors::Result<()> {
...
+            meta.shard_footer.stored_bytes_on_disk
+        } else {
+            meta.shard_footer.stored_bytes
+        };
         materialized_bytes += meta.shard_footer.materialized_bytes;
         stored_bytes += meta.shard_footer.stored_bytes;
     }
 
     println!("{{");
+    println!("\"total_compressed_cas_bytes\" : {stored_bytes_on_disk},");
```
Comment (port8080) [inline rust/gitxetcore/src/data/mdb.rs:909]: nit: maybe only print this line if it is different from stored_bytes?

**Review by port8080 — APPROVED** [review]

File: rust/mdb_shard/src/shard_format.rs [hunk 2]
```diff
@@ -234,8 +238,13 @@ impl MDBShardInfo {
...
 
-        // Move cursor to end of shard file minus footer size.
-        reader.seek(SeekFrom::End(-MDB_SHARD_FOOTER_SIZE))?;
+        // Move cursor to the end of shard file minus sizeof(u64) to
+        // read footer_offset. Then seek to that position to read
+        // MDBShardFileFooter.
+        //
+        reader.seek(SeekFrom::End(-(size_of::<u64>() as i64)))?;
+        let footer_offset = read_u64(reader)?;
+        reader.seek(SeekFrom::Start(footer_offset))?;
```
[the remark this answers is not in the document]
Comment (hoytak) [reply rust/mdb_shard/src/shard_format.rs:247]: I think it's for sure backwards compatible for now.  (We can add an additional section of config info if more sections are needed too)

**Review by sirahd — APPROVED** [review]
🚀

**Review by hoytak — APPROVED** [review]
LGTM!

File: rust/gitxetcore/src/data/mdb.rs (hunk 1 above)
Comment (seanses) [reply rust/gitxetcore/src/data/mdb.rs]: Done

**Pull request merged**